overlord/snapstate/snapstate_install_test.go

```diff
@@ -6018,3 +6018,45 @@ func (s *snapmgrTestSuite) TestInstallPathManySplitEssentialWithoutSharedBased(c
 	c.Check(chg.IsReady(), Equals, true)
 	c.Check(chg.Status(), Equals, state.DoneStatus)
 }
+
+func (s *snapmgrTestSuite) TestInstallManyNoSnaps(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// try to install an empty list of snaps
+	_, _, err := snapstate.InstallMany(s.state, nil, nil, s.user.ID, nil)
+	c.Assert(err, NotNil)
```

Comment:
If you're using ErrorMatches, I don't think you need this one. That checker already handles a nil result gracefully